File: frontend/services/flashcards.ts
export async function generateFlashcards(text: string) {
  const response = await fetch(
    "http://127.0.0.1:8000/generate-flashcards",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        text,
      }),
    }
  );

  if (!response.ok) {
    throw new Error("Failed to generate flashcards");
  }

  return response.json();
}

File: backend/app/routes/flashcards.py
from fastapi import APIRouter, HTTPException

from app.schemas.flashcards import FlashcardRequest
from app.services.flashcards_service import generate_flashcards

router = APIRouter()


@router.post("/generate-flashcards")
def flashcards(request: FlashcardRequest):

    try:

        if not request.text.strip():
            raise HTTPException(
                status_code=400,
                detail="Study material cannot be empty."
            )

        flashcards = generate_flashcards(request.text)

        if isinstance(flashcards, dict) and "error" in flashcards:
            raise HTTPException(
                status_code=500,
                detail=flashcards["error"]
            )

        return {
            "success": True,
            "flashcards": flashcards
        }

    except HTTPException:
        raise

    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Failed to generate flashcards: {str(e)}"
        )